source/simulator/newton.cc

```diff
@@ -331,6 +339,13 @@ namespace aspect
                            "When this parameter is true and the linear solver fails, we try again, but now with SPD stabilization "
                            "for both the preconditioner and the velocity block. The SPD stabilization will remain active untill "
                            "the next timestep, when the default values are restored.");
+
+
+        prm.declare_entry ("SPD safety factor", "0.9",
+                           Patterns::Double (0,1),
+                           "The SPD factor ensures positive semi-definiteness. So when the the SPD factor is smaller then one, it "
+                           "scales the eigenvalue back to zero. In practice we want to stay away from a zero eigenvalue, so we scale "
+                           "the SPD value by this value.");
```

Comment:
Write this as follows:
```
  When stabilizing the Newton matrix, we can encounter situations where the coefficient inside the elliptic (top-left)
  block becomes negative or zero. This coefficient has the form $1+x$ where $x$ can sometimes be smaller than
  $-1$. In this case, the top-left block of the matrix is no longer positive definite, and both preconditioners and
  iterative solvers may fail. To prevent this, the stabilization computes an $\alpha$ so that $1+\alpha x$ is
  never negative. This $\alpha$ is chosen as $1$ if $x\ge -1$, and $\alpha=-\frac 1x$ otherwise. (Note that
  this always leads to $0\le \alpha \le 1$.)  On the
  other hand, we also want to stay away from $1+\alpha x=0$, and so modify the choice of $\alpha$ to
  be $1$ if $x\ge -c$, and $\alpha=-\frac cx$ with a $c$ between zero and one. This way, if $c<1$, we are
  assured that $1-\alpha x>c$, i.e., bounded away from zero.
```